### ISSUE TITLE
[appName] won't run without Google Play services, which are not supported by your device

### ISSUE THREAD

**Issue by H1Gerd** [microg/android_packages_apps_GmsCore#831]
I installed micro g using nanodroid. every relevant self check works, including the signature spoof checker.

However I cant even use youtube - where my 2nd phone running on microg too doesnt have this problem. Opening the app will return "youtube won't run without Google Play services, which are not supported by your device". This problem also appears with other apps (that also work on my 2nd phone)
I also installed gsync for calender and contacts - however these doesnt work as well

Im Using Carbon Rom, which supports spoofing based on your signature spoofing entry (https://github.com/microg/android_packages_apps_GmsCore/wiki/Signature-Spoofing)

I reflashed the complete rom, however i still have the same problem

**Comment by Mynacol** [comment]
Instead of using the official YouTube app, you could switch to [NewPipe](https://f-droid.org/app/org.schabi.newpipe), a very powerful unofficial YouTube client, which doesn't need Google Services on the device.

**Comment by Nanolx** [comment]
Everything's working fine here, so I suppose there's something wrong on your setup. Either way, without logcat we can't help you.

**Comment by H1Gerd** [comment]
Thx for the fast replies.

So in the device Logs i get the warning
    GoogePlayServicesUtil: Google Play Store signature Invalid
followed by  the error
   GoogePlayServicesUtil: GooglePlayServices not availble due to error 9
I think these are the errors

See:
[logcat.txt](https://github.com/microg/android_packages_apps_GmsCore/files/3269420/logcat.txt)

**Comment by Nanolx** [comment]
Did you grant Play Store the Signature Spoofing permission?

**Comment by H1Gerd** [comment]
I don't have any gapps, including the playstore, installed

The onlay app with signature spoofing permission is the microG services Core

**Comment by Nanolx** [comment]
Aha, then I know the what's the cause: you need to install Fake Store and grant it Signature Spoofing permission. Fake Store is a pseudo app part of the microG ecosystem which mocks the existence of Play Store and thus solves your issue.

**Comment by H1Gerd** [comment]
That makes sense. However I just added the microg repo for fdroid (which installed the fake store) and I still get the error :confused:

**Comment by Nanolx** [comment]
Did you grant the permisson? Enabling device registration in microG Settings (and a reboot afterwards) might be required, aswell.

**Comment by ale5000-git** [comment]
@H1Gerd: I don't remember but it is possible that Fake Store doesn't ask for Signature Spoofing permission, but you still have to grant it.
Look in the Android settings.

**Comment by Nanolx** [comment]
In some ROMs you need to go to Signature Spoofing Permssions > 3 dot menu > Show all apps to see Fake Store. In some other ROMs only /system applications can be granted Signature Spoofing permission.

Since you've already used NanoDroid, you may configure your setup to let it install Fake Store instead of nothing.

**Comment by H1Gerd** [comment]
> 
> 
> @H1Gerd: I don't remember but it is possible that Fake Store doesn't ask for Signature Spoofing permission itself, but you still have to grant it.
> Look in the Android settings.

I had to to click show system in the settings of the spoof permissions - than the fakestore appeared :man_facepalming: 
sry this was so stupid from me

Thanks for helping @Nanolx and @ale5000-git - The problem is solved now
And also ty for the NewPipe suggestion - I thought youtube vanced is they shit but NewPipe beats it

**Comment by H1Gerd** [comment]
> 
> 
> In some ROMs you need to go to Signature Spoofing Permssions > 3 dot menu > Show all apps to see Fake Store. In some other ROMs only /system applications can be granted Signature Spoofing permission.
> 
> Since you've already used NanoDroid, you may configure your setup to let it install Fake Store instead of nothing.

One question - i read through your git docu. You wrote: "... use the provided build-package"
However i didnt found this pckg. Am i to stupid to find it or is this just working with unix based os (linux or macos)?

**Comment by Nanolx** [comment]
`build-package` is a script in the toplevel directory of the NanoDroid source repository.

As you can see the repository is mostly empty. The `build-package` script is used - as described to

a) download all required APK and libraries for distribution
b) create the desired NanoDroid package

Yes, `build-package` only works on unixoid operating systems (though, it may work with Cygwin on Windows, but I never tested that).

**Comment by H1Gerd** [comment]
Good to know. Maybe I'll give it a try next time.
Thx and sry for stealing your time

Have a nice weekend!

**Comment by ndgnuh** [comment]
On some ROM, the menu doesn't show up.

Here's the solution in that case: https://shadow53.com/android/no-gapps/faq/playstore-incorrect-signature/
> If you use one of my microG zips or NanoMod’s microG zip, you may find that microG reports the Play Store as not having the right signature.
> 
> If you are using a zip with the Play Store, this is because the author of NanoMod has patched the Play Store apk to prevent the app from crashing when you try to make purchases on Google Play. If you are using FakeStore, it’s because you aren’t using the Play Store.
> 
> To fix the issue for both apps, open a terminal on your phone or enter adb shell on a computer and enter the following commands to gain root access and manually grant the signature spoofing permission:
> 
> su
> pm grant com.android.vending android.permission.FAKE_PACKAGE_SIGNATURE
> 
> Close the Play Store if it is open and exit the microG “Self-Check” (if you haven’t already), then view the Self-Check page again. It should report that the Play Store has the correct signature.


Just posting this in case someone run into the same problem as I am.

**Comment by thomas725** [comment]
I'm unsure if my problem is related. I wanted to try the game "Diablo Immortal" on my microg phone, but it crashes on trying to login / start game as guest.

I found this error in logcat:
> ActivityManager: Unable to start service Intent { act=com.google.android.play.core.assetmoduleservice.BIND_ASSET_MODULE_SERVICE pkg=com.android.vending } U=0: not found

I suspected something's wrong with my fake store. I've checked the self-check output of microg settings, and everything looks fine there. My ROM (linageos microg android 11) doesn't have the setting talked about above, and trying the root shell command suggested by @ndgnuh gives me this error:
> java.lang.SecurityException: Permission android.permission.FAKE_PACKAGE_SIGNATURE requested by com.android.vending is not a changeable permission type

Does anybody have any hint's what else I could try?
